examples/all-clusters-app/all-clusters-common/include/EnergyEvseDelegateImpl.h

```diff
@@ -0,0 +1,193 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "app/clusters/energy-evse-server/energy-evse-server.h"
+
+#include <app/util/af.h>
+#include <app/util/config.h>
+#include <cstring>
+
+using chip::Protocols::InteractionModel::Status;
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace EnergyEvse {
+
+/**
+ * The application delegate.
+ */
+
+class EnergyEvseDelegate : public EnergyEvse::Delegate
+{
+public:
+    /**
+     * @brief   Called when EVSE cluster receives Disable command
+     */
+    Status Disable() override;
+
+    /**
+     * @brief   Called when EVSE cluster receives EnableCharging command
+     *
+     * @param chargingEnabledUntil
+     * @param minimumChargeCurrent (in mA)
+     * @param maximumChargeCurrent (in mA)
+     */
+    Status EnableCharging(const chip::app::DataModel::Nullable<uint32_t> & chargingEnabledUntil,
+                          const int64_t & minimumChargeCurrent, const int64_t & maximumChargeCurrent) override;
+
+    /**
+     * @brief   Called when EVSE cluster receives EnableDischarging command
+     *
+     * @param dischargingEnabledUntil
+     * @param maximumChargeCurrent (in mA)
+     */
+    Status EnableDischarging(const chip::app::DataModel::Nullable<uint32_t> & dischargingEnabledUntil,
+                             const int64_t & maximumDischargeCurrent) override;
+
+    /**
+     * @brief    Called when EVSE cluster receives StartDiagnostics command
+     */
+    Status StartDiagnostics() override;
+
+    // -----------------------------------------------------------------
+    // Internal API to allow an EVSE to change its internal state etc
+    // TODO Status hwRegisterEvseHardwareCallback(Callback);
+    Status hwSetMaxHardwareCurrentLimit(int64_t currentmA);
+    Status hwSetCircuitCapacity(int64_t currentmA);
+    Status hwSetCableAssemblyLimit(int64_t currentmA);
+    Status hwSetState(StateEnum state);
+    Status hwSetFault(FaultStateEnum fault);
```

Comment:
We generally use InitialCaps naming for methods....

Reply:
Fixed